### PR TITLE
Merge elemental installer

### PR DESCRIPTION
**Pull request by davidcassany**
This PR includes the former `ros-installer` code in this repository. `ros-installer` is then called `elemental-installer`.

Related to #7

### CODE REVIEW

**Comment by kkaempf** [comment]
Will you do the "rancheros" -> "elemental" renames in this PR or a separate one ?

**Comment by kkaempf** [comment]
https://build.opensuse.org/package/show/isv:Rancher:Elemental/elemental-operator is adapted and building fine ;-)

**Comment by davidcassany** [comment]
> Will you do the "rancheros" -> "elemental" renames in this PR or a separate one ?

Gonna do it here, I was trying to see of if there was some way from `elemental` repository to consume this right away and attempt some manual checks, but probably is faster going through a regular build using the OSB image.

**Comment by davidcassany** [comment]
I consider this ready to be merged. It is not actually tested yet, it is just a port form `ros-installer`, it builds and former ros-installer tests are all passing. Changes in former elemental-operator are really cosmetic, just a little project re-structure to move former `main.go` into a subfolder.

File: .github/workflows/unit-tests.yaml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+name: Unit tests
```
Comment (Itxaka) [inline .github/workflows/unit-tests.yaml:1]: whats the difference between this one and the .github/workflows/tests.yaml?

File: .github/workflows/unit-tests.yaml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+        go-version: '^1.16'
+    - name: Get dependencies
+      run: |
+        # Needed for github.com/google/go-tspi/tspi
+        # in opensuse this is trousers-devel
+        sudo apt-get install libtspi-dev
+        make unit-tests-deps
+    - name: Run tests
+      run: |
+        make unit-tests
```
Comment (Itxaka) [inline .github/workflows/unit-tests.yaml:24]:
```suggestion
        make unit-tests
    - name: Codecov
      uses: codecov/codecov-action@v2
      with:
        file: ./coverage.out
```

**Review by Itxaka — CHANGES_REQUESTED** [review]
This requires changes to the goreleaser so it can build both binaries.
also to the Dockerfile so it picks the newer binary

File: .github/workflows/unit-tests.yaml (hunk 1 above)
Comment (davidcassany) [reply .github/workflows/unit-tests.yaml:1]: This one is the one coming from the `ros-installer`. I realized both tests are not colliding and the former tests.yaml from `elemental-operator` is broken. Hence I did not want to mess with that in this PR, simply add the new logic without altering whatever it was already there.

Comment (Itxaka) [reply .github/workflows/unit-tests.yaml:1]: sounds good, I will merge them together on my PR :+1:

**Review by Itxaka — APPROVED** [review]
Not sure what I wrote here lol

**Comment by Itxaka** [comment]
lint,unit and e2e tests are being fixed in a different PR so it should not be a blocker to merge this.

Also we should consider following up on this with a config for the coverage to treat the installer and the operator as different packages, as Im pretty sure the installer has a 3% coverage or something like that lol

**Comment by Itxaka** [comment]
@davidcassany nah, Im still working on that one, will take some time still so lets merge this, I will rebase and fix on my side!

**Comment by davidcassany** [comment]
> We should probably tag it with a version > 0.1.0 (last ros-installer release) !?

yes makes sense, I think we could tag something like 0.1.y and then tag  v0.2.x once we have all the renaming and smoke tests in place.

**Pull request merged**